sw/device/tests/rv_core_ibex_rnd_test.c

```diff
@@ -0,0 +1,99 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/lib/base/memory.h"
+#include "sw/device/lib/base/mmio.h"
+#include "sw/device/lib/dif/dif_edn.h"
+#include "sw/device/lib/dif/dif_rv_core_ibex.h"
+#include "sw/device/lib/runtime/log.h"
+#include "sw/device/lib/testing/entropy_testutils.h"
+#include "sw/device/lib/testing/test_framework/check.h"
+#include "sw/device/lib/testing/test_framework/ottf_main.h"
+
+#include "hw/top_earlgrey/sw/autogen/top_earlgrey.h"
+#include "rv_core_ibex_regs.h"
+
+OTTF_DEFINE_TEST_CONFIG();
+
+dif_result_t rv_core_ibex_read_rnd_data_followed_by_status(
+    const dif_rv_core_ibex_t *rv_core_ibex, uint32_t *data,
+    dif_rv_core_ibex_rnd_status_t *followed_status) {
+  if (rv_core_ibex == NULL || data == NULL || followed_status == NULL) {
+    return kDifBadArg;
+  }
+  uint32_t reg = mmio_region_read32(rv_core_ibex->base_addr,
+                                    RV_CORE_IBEX_RND_STATUS_REG_OFFSET);
+  if (!bitfield_bit32_read(reg, RV_CORE_IBEX_RND_STATUS_RND_DATA_VALID_BIT)) {
+    return kDifError;
+  }
+
+  *data = mmio_region_read32(rv_core_ibex->base_addr,
+                             RV_CORE_IBEX_RND_DATA_REG_OFFSET);
+
+  // There are only 10 cycles between reading RND and status returning back
+  // high.
+  *followed_status = mmio_region_read32(rv_core_ibex->base_addr,
+                                        RV_CORE_IBEX_RND_STATUS_REG_OFFSET);
+  return kDifOk;
+}
```

Comment:
I wonder if this (and the method below) are needed? Perhaps we could just expand this inline (invoking DIFs directly) in `test_main()`?

It is unexpected to see test code directly interact with low level device registers - it is preferred that tests only invoke DIFs or the shared testutils (which is one layer above DIFs). If you think there is scope for sharing these methods with other tests, then I suggest adding `sw/device/lib/testing/rv_core_ibex_testutils.c|h`.